lib/Controller/Schedule.php

```diff
@@ -1727,8 +1727,23 @@ public function edit(Request $request, Response $response, $id)
         $schedule->campaignId = $this->isFullScreenSchedule($schedule->eventTypeId)
             ? $sanitizedParams->getInt('fullScreenCampaignId')
             : $sanitizedParams->getInt('campaignId');
-        $schedule->displayOrder = $sanitizedParams->getInt('displayOrder', ['default' => 0]);
-        $schedule->isPriority = $sanitizedParams->getInt('isPriority', ['default' => 0]);
+
+        // displayOrder and isPriority: if present but empty (""): set to 0
+        // if missing from form: keep existing value (fallback to 0 if unset)
+        $params = $request->getParams();
+
+        $schedule->displayOrder = array_key_exists('displayOrder', $params)
```

Comment:
Did you try `$sanitizedParams->hasParam()` ?

Reply:
> Did you try `$sanitizedParams->hasParam()` ?

Didn't know that method existed, that simplifies things a lot, I've made a new commit, thank you!